lib/aws/AwsSigv4Signer.js

```diff
@@ -77,13 +78,23 @@ function AwsSigv4Signer(opts = {}) {
     request.region = opts.region;
     request.headers = request.headers || {};
     request.headers['host'] = request.hostname;
+    request.headers['X-Amz-Content-Sha256'] = hash(request.body || '', 'hex');
```

Comment:
Looks like this was a bit more involved, https://github.com/opensearch-project/opensearch-js/pull/366